### PR TITLE
Add Release Workflow to OSB

### PR DESCRIPTION
**Pull request by IanHoang**
### Description
This PR achieves the following:
- A GitHub action workflow that will be triggered when a tag is pushed to this repository. The workflow builds the product and creates a draft release with built artifacts attached as artifacts.tar.gz in the release.
- A jenkins workflow that is triggered based on above draft release. The workflow looks for artifacts.tar.gz, downloads it and then signs it and publishes it to [PyPi](https://pypi.org/project/opensearch-benchmark/)

### Issues Resolved
[List any issues this PR will resolve]

### Testing
- [ ] New functionality includes testing

[Describe how this change was tested]

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RELEASING.md
Comment (gaiksaya) [inline RELEASING.md:5]: You can below link here:
https://build.ci.opensearch.org/job/opensearch-benchmark-release

File: .github/workflows/release-drafter.yml
Comment (IanHoang) [reply .github/workflows/release-drafter.yml:34]: Will give this a shot

Comment (IanHoang) [reply .github/workflows/release-drafter.yml:34]: Ran `python -m pip install --upgrade build` and `python -m build` successfully:
```
# Output of second command:
Successfully built opensearch-benchmark-0.1.0.tar.gz and opensearch_benchmark-0.1.0-py3-none-any.whl

# dist/ directory 
hoangia@3c22fbd0d988 opensearch-benchmark % ls dist
opensearch-benchmark-0.1.1.tar.gz		opensearch_benchmark-0.1.1-py3-none-any.whl
```

File: .github/workflows/release-drafter.yml
Comment (gaiksaya) [inline .github/workflows/release-drafter.yml:21]: Make this as 1 since the number of code owners in this repo are just 2 and one of them will be the release manager. https://github.com/opensearch-project/opensearch-benchmark/blob/main/.github/CODEOWNERS

Comment (IanHoang) [reply .github/workflows/release-drafter.yml]: Done!

File: version.txt
Comment (gaiksaya) [inline version.txt:1]: Is this just a patch release?

**Review by gaiksaya — APPROVED** [review]
The test failures are not related to this change. Would recommend squash the 10 commits to single clean commit before merging.

File: version.txt
Comment (IanHoang) [reply version.txt:1]: Oops, I believe it's a minor release since it includes minor bug fixes and features that are backwards compatible

Comment (rishabh6788) [reply version.txt:1]: LGTM, please bump the minor version. Approving.

**Review by rishabh6788 — APPROVED** [review]

**Pull request merged**
